### ISSUE TITLE
Goblint crashes on `ls_comb.c` on library function

### ISSUE THREAD

**Issue by jerhard** [goblint/analyzer#947]
When the current master is run on `ls-comb` from the `bench` repository, it crashes during analysis with:
```
Fatal error: exception Invalid_argument("accs")
``` 
Which is raised in `LibraryDsl`.

Command line:
```
./goblint ../bench/coreutils/ls_comb.c --set pre.cppflags[+] "--std=gnu89"
```

The line that is analyzed when the crash occurs is:

```
#line 1361
    _obstack_begin(& dev_ino_obstack, 0, 0, (void *(*)(long  ))(& malloc), & free);
```
Where `_obstack_begin` is [some library function](https://refspecs.linuxfoundation.org/LSB_1.0.0/gLSB/baselib--obstack-begin.html). 

<details>
<summary> Stack trace </summary>

```
[Error][Analyzer] About to crash at mark transfer function at ls.c:1361:5-1361:82!

runtime: 00:00:09.719
vars: 21631, evals: 66998

|rho|=21631
|called|=28
|stable|=19731
|infl|=21410
|wpoint|=24
|side_dep|=0
|side_infl|=0
|rho_write|=0
|dep|=19802
Found 249 contexts for 84 functions. Top 5 functions:
41	contexts for entry state of quotearg_buffer_restyled (20513) on quotearg.c:202:1-613:1
13	contexts for entry state of quotearg_n_options (20518) on quotearg.c:719:1-777:1
10	contexts for entry state of argmatch (3719) on argmatch.c:82:1-124:1
10	contexts for entry state of quotearg_n_style (4456) on quotearg.c:803:1-809:1
8	contexts for entry state of quote_n (4461) on quote.c:28:1-33:1

Memory statistics: total=19312.72MB, max=200.75MB, minor=19300.62MB, major=277.17MB, promoted=265.07MB
    minor collections=9210  major collections=12 compactions=0


vfprintf.c:45: Warning: Invalid position for (prefix) function type attributes: __attribute__((__nonnull__(1,4))) 
vfprintf.c:54: Warning: Invalid position for (prefix) function type attributes: __attribute__((__nonnull__(1,4))) 
vfprintf.c:54: Warning: Invalid position for (prefix) function type attributes: __attribute__((__nonnull__(1,4))) 
fd-reopen.c:33: Warning: Invalid position for (prefix) function type attributes: __attribute__((__artificial__)) 
:-1: Warning: Incompatible declaration for pthread_mutex_init (from .goblint/preprocessed/ls_comb.i(2)).
 Previous was at /usr/include/pthread.h:781 (from .goblint/preprocessed/pthread.i (1)) 
	Failed assumption that pthread_mutex_t and pthread_mutex_t are isomorphic 
	Failed assumption that union __anonunion_pthread_mutex_t_335460617 and union __anonunion_pthread_mutex_t_11 are isomorphic (different number of fields in __pthread_mutex_s and __pthread_mutex_s: 8 != 7.)
union __anonunion_pthread_mutex_t_335460617 {
   struct __pthread_mutex_s __data ;
   char __size[40] ;
   long __align ;
};

union __anonunion_pthread_mutex_t_11 {
   struct __pthread_mutex_s __data ;
   char __size[40] ;
   long __align ;
};
 
Fatal error: exception Invalid_argument("accs")
```
</details>

**Comment by sim642** [comment]
I fixed it directly on master because the root issue was quite simple.

The problem was that calling the library function `_obstack_begin` spawns its arguments as threads as well. But its arguments are `malloc` and `free`, which are both also library functions and take arguments. However, when spawning threads from special functions, an empty argument list was used that doesn't match with `malloc` and `free`'s library function specification.
